src/connectionhealthpolicy/SendingAudioFailureConnectionHealthPolicy.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Logger from '../logger/Logger';
+import BaseConnectionHealthPolicy from './BaseConnectionHealthPolicy';
+import ConnectionHealthData from './ConnectionHealthData';
+import ConnectionHealthPolicy from './ConnectionHealthPolicy';
+import ConnectionHealthPolicyConfiguration from './ConnectionHealthPolicyConfiguration';
+
+export default class SendingAudioFailureConnectionHealthPolicy
+  extends BaseConnectionHealthPolicy
+  implements ConnectionHealthPolicy {
+  private readonly secondsToConsider: number;
+  private readonly initialWaitTimeMs: number;
+  private readonly coolDownTimeMs: number;
+  private readonly maximumTimesToWarn: number;
+  private warnCount: number;
+  private lastWarnTimestampMs: number;
+
+  constructor(
+    private logger: Logger,
+    configuration: ConnectionHealthPolicyConfiguration,
+    data: ConnectionHealthData
+  ) {
+    super(configuration, data, 'Sending Audio Health');
+    this.secondsToConsider = configuration.sendingAudioFailureSamplesToConsider;
+    this.initialWaitTimeMs = configuration.sendingAudioFailureInitialWaitTimeMs;
+    this.maximumTimesToWarn = configuration.maximumTimesToWarn;
+    this.coolDownTimeMs = configuration.cooldownTimeMs;
+    this.lastWarnTimestampMs = 0;
+    this.warnCount = 0;
+  }
+
+  private isSendingAudioUnHealthy(): boolean {
+    const hasEnoughTimeElapsedToEvaluateStatus = !this.currentData.isConnectionStartRecent(
+      this.initialWaitTimeMs
+    );
+    const areAudioPacketsNotBeingSent =
+      this.currentData.consecutiveStatsWithNoAudioPacketsSent >= this.secondsToConsider;
```

Comment:
Can we use `configuration.sendingAudioFailureSamplesToConsider` over `this.secondsToConsider` and just drop that variable? I am a bit confused one maps to seconds and one maps to sample count.

Reply:
Yeah, I admit it's a bit confusing, I think just using `sendingAudioFailureSamplesToConsider` everywhere is clear.

 PS: The reason I used `secondsToConsider` is because we only collect RTC stats once every second